### PR TITLE
Deprecate `proc` kwarg to call_historic()

### LINKED ISSUE #120: Change call_historic callback kwarg name
As per discussion in #110 we agreed that the kwarg `proc` in `_HookCaller.call_historic()` is poorly named and should be instead renamed `result_callback`. 

`proc` should be deprecated in a minor release with an appropriate warning and later removed.

### PR DESCRIPTION
**Pull request by tgoodlet**
It's a bad name, use `result_callback` instead.

Resolves #120

### CODE REVIEW

File: pluggy/hooks.py [hunk 1]
```diff
@@ -257,20 +257,29 @@ def __call__(self, *args, **kwargs):
...
-    def call_historic(self, proc=None, kwargs=None):
-        """ call the hook with given ``kwargs`` for all registered plugins and
+    def call_historic(self, result_callback=None, kwargs=None, proc=None):
+        """Call the hook with given ``kwargs`` for all registered plugins and
         for all plugins which will be registered afterwards.
 
-        If ``proc`` is not None it will be called for for each non-None result
-        obtained from a hook implementation.
+        If ``result_callback`` is not ``None`` it will be called for for each
+        non-None result obtained from a hook implementation.
```
Comment (nicoddemus) [inline pluggy/hooks.py]: Please document that `proc` is deprecated in the docstring as well.

File: testing/test_pluginmanager.py [hunk 2]
```diff
@@ -217,7 +217,13 @@ def he_method1(self, arg):
     pm.register(Plugin1())
 
     he_method1 = pm.hook.he_method1
-    he_method1.call_historic(proc=callback, kwargs=dict(arg=1))
+
+    # TODO: remove once ``proc`` is removed
+    if callback:
```
Comment (nicoddemus) [inline testing/test_pluginmanager.py]: For completeness we should also test passing `result_callback`. I suggest changing this test to use `result_callback` and create a new, simpler test, just for the deprecation.

File: pluggy/hooks.py (hunk 1 above)
Comment (tgoodlet) [reply pluggy/hooks.py]: Will do!

File: testing/test_pluginmanager.py (hunk 2 above)
Comment (tgoodlet) [reply testing/test_pluginmanager.py]: Sure thing!

**Comment by tgoodlet** [comment]
@nicoddemus pushed all your recs.

File: pluggy/hooks.py [hunk 3]
```diff
@@ -257,20 +257,32 @@ def __call__(self, *args, **kwargs):
...
+            )
+            result_callback = proc
+
+        self._call_history.append((kwargs or {}, result_callback))
         # historizing hooks don't return results
         res = self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
-        if proc is None:
+        if result_callback is None:
             return
+        # XXX: woah isn't this wrong for `firstresult` hooks?
```
Comment (tgoodlet) [inline pluggy/hooks.py]: @nicoddemus btw did you notice this?
If i'm reading this correctly won't this code never process `firstresult` hooks?

Comment (tgoodlet) [reply pluggy/hooks.py]: Yeah I think that loop expr is just slightly wrong should be `for x in res or [res]:` right?

File: pluggy/hooks.py [hunk 4]
```diff
@@ -257,20 +257,32 @@ def __call__(self, *args, **kwargs):
...
+        If ``result_callback`` is not ``None`` it will be called for for each
+        non-None result obtained from a hook implementation.
+
+        .. note::
+            The ``proc`` argument is now deprecated.
         """
-        self._call_history.append((kwargs or {}, proc))
+        if proc is not None:
+            warnings.warn(
+                "Support for `proc` argument now deprecated and will be"
```
Comment (tgoodlet) [inline pluggy/hooks.py]: `Support for `proc` **is** argument now`  dahhh!

File: pluggy/hooks.py (hunk 3 above)
Comment (tgoodlet) [reply pluggy/hooks.py]: Oh waitttt [whatttt](https://github.com/pytest-dev/pluggy/blob/master/pluggy/hooks.py#L35)?
Wait why not?

**Comment by tgoodlet** [comment]
@nicoddemus @RonnyPfannschmidt give this one more lookover if you don't mind. I tweaked couple things and added docs.

**Pull request merged**